tests/unit/test_experiments/test_experiment.py

```diff
@@ -82,6 +82,24 @@ def test_read_strings_repeat(self):
         )
         self.assertEqual(experiment.period, 60)
 
+    def test_cycle_unpacking(self):
+        experiment = pybamm.Experiment(
+            [
+                ("Discharge at C/20 for 0.5 hours", "Charge at C/5 for 45 minutes"),
+                ("Discharge at C/20 for 0.5 hours", "Charge at C/5 for 45 minutes"),
+            ]
+        )
+        self.assertEqual(
+            experiment.operating_conditions,
+            [
+                (0.05, "C", 1800.0, 60.0),
+                (-0.2, "C", 2700.0, 60.0),
+                (0.05, "C", 1800.0, 60.0),
+                (-0.2, "C", 2700.0, 60.0),
+            ],
+        )
+        self.assertEqual(experiment.cycle_lengths, [2, 2])
```

Comment:
add test for cycles of different lengths, and mixed cycle/string (see comment above)